### DISCUSSION TITLE
Is model parallelism supported for PyTorch?

### DISCUSSION THREAD

**Discussion by liaopeiyuan** [horovod/horovod#2837] [Ideas]
If I write my own multi-GPU model or use `torch.distributed.pipeline.sync.Pipe`, would multi-node training still work with horovod?

**Comment by tgaddair** [comment]
Hey @liaopeiyuan, Horovod can still work in model parallel scenarios. The important thing to watch out for is that every Horovod worker needs to submit the same set of tensors as the others. So we usually implement model parallelism by having the parallelism branch out from the Horovod worker (for example, each Horovod worker gets 2 GPUs).

**Reply by dinhanhx** [reply]
Uhm can you kindly give example?

**Reply by maxhgerlach** [reply]
Have a look at this unit test: https://github.com/horovod/horovod/blob/40cbc9b4820c278e1ecd75eaf88c7aeaf3c61a62/test/parallel/test_torch.py#L1860

**Reply by dinhanhx** [reply]
Thank you, @maxhgerlach
